packages/compartment-mapper/src/link.js

```diff
@@ -351,7 +353,11 @@ export const link = (
       parserForLanguage,
       moduleTransforms,
     );
-    const importHook = makeImportHook(location, name, parse);
+    /** @type {ShouldDeferError} */
+    const shouldDeferError = language =>
+      language && parserForLanguage[language].heuristicImports;
```

Comment:
Please add a check for whether `has(parserForLanguage, language)` and an appropriate behavior for the case the language is missing.

Reply:
but typescript said it was fine! xD

done